rclpy_examples/talker_asyncio.py

```diff
@@ -0,0 +1,33 @@
+import asyncio
+import sys
+
+import rclpy
+from rclpy.qos import qos_profile_default
+from std_msgs.msg import String
+
+
+@asincio.coroutine
+def main(args):
+    rclpy.init(args)
+
+    node = rclpy.create_node("talker")
+
+    chatter_pub = node.create_publisher(String, "chatter", qos_profile_default)
+
+    loop_rate = rclpy.WallRate(2)
+
+    msg = String()
+
+    i = 1
+    while rclpy.ok():
+      msg.data = "Hello World: " + i
+      i += 1
+      print("Publishing: '%s'" % msg.data)
+      chatter_pub.publish(msg)
+      yield from rclpy.spin_some(node)
+      yield from loop_rate.sleep()
```

Comment:
Another way to write it:

```Python
import asyncio
import sys

import rclpy
from std_msgs.msg import String


@asyncio.coroutine
def publish_loop(publisher, loop_rate):
    msg = String()
    i = 1
    while rclpy.ok():
        msg.data = "Hello World: {0}".format(i)
        i += 1
        print("Publishing: '{0}'".format(msg.data))
        publisher.publish(msg)
        yield from loop_rate.sleep()


def main(args):
    rclpy.init(args)
    node = rclpy.create_node("talker")
    chatter_pub = node.create_publisher(String, "chatter", rclpy.qos_profile_default)
    loop_rate = rclpy.WallRate(2)
    asyncio.ensure_future(publish_loop(chatter_pub, loop_rate))
    loop = asyncio.get_event_loop()
    asyncio.ensure_future(rclpy.spin(node, on_shutdown=lambda: loop.close()))
    loop.run_forever()

if __name__ == '__main__':
    main(sys.argv)

```

Which could be simplified further with some syntactic sugar from `rclpy`:

```Python
import asyncio
import sys

import rclpy
from std_msgs.msg import String


@asyncio.coroutine
def publish_loop(publisher, loop_rate):
    msg = String()
    i = 1
    while rclpy.ok():
        msg.data = "Hello World: {0}".format(i)
        i += 1
        print("Publishing: '{0}'".format(msg.data))
        publisher.publish(msg)
        yield from loop_rate.sleep()


def main(args):
    rclpy.init(args)
    node = rclpy.create_node("talker")
    chatter_pub = node.create_publisher(String, "chatter", rclpy.qos_profile_default)
    loop_rate = rclpy.WallRate(2)
    asyncio.ensure_future(publish_loop(chatter_pub, loop_rate))
    rclpy.asyncio_spin(node, loop=asyncio.get_event_loop())

if __name__ == '__main__':
    main(sys.argv)

```